dds/DCPS/RTPS/TypeLookup.h

```diff
@@ -82,7 +82,7 @@ struct TypeLookup_getTypeDependencies_Result { //IDL: union switch (long)
 };
 
 struct TypeLookup_Call { //IDL: union switch (long)
-  ACE_CDR::Long kind;
+  ACE_CDR::ULong kind;
```

Comment:
It was that way to match the IDL in 7.6.3.3.3 in the spec.

Reply:
I did it because I got this warning:

```
Sedp.cpp: In member function ‘DDS::ReturnCode_t OpenDDS::RTPS::Sedp::TypeLookupReplyReader::process_type_lookup_reply(const OpenDDS::DCPS::ReceivedDataSample&, OpenDDS::DCPS::Serializer&, bool)’:
Sedp.cpp:4031: warning: comparison between signed and unsigned integer expressions
```

https://github.com/objectcomputing/OpenDDS/blob/2168fc3ac3ca1acea8e6f4b162baea1e3ab2254b/dds/DCPS/RTPS/Sedp.cpp#L4031

I assumed it was a typo on our part. Since it's in the spec, then the spec is contradicting itself. It declares the id constants as `unsigned long`s, but the union switch type is `long`. Unsigned would be correct since they're member ids.

Reply:
It's signed in the RPC spec, so I assume that's where it came from.
I guess I'd rather have the constants match the spec and use a cast in the code.

Reply:
I'd rather fix the spec if it's contradicting it self. I don't really know anything about the RPC spec, but if it's supposed to be `long`, then it sounds like the constants have to change.

Reply:
I agree with both of you.  I recommend:

1. The IDL should match the spec.
2. Create a ticket against the spec for the contradiction.
3. Put cast in code to silence warning.
4. Create internal ticket to revisit spec in future.

  Can we create an issue for the spec to work out the

Reply:
I'd be fine with using a cast as long as an issue is filed.